backend/api/pretix/mutations.py

```diff
@@ -0,0 +1,65 @@
+import logging
+
+import strawberry
+from strawberry.types import Info
+
+import pretix
+from api.permissions import IsAuthenticated
+from api.pretix.query import get_user_tickets
+from api.pretix.types import AttendeeTicket, UpdateAttendeeTicketInput
+from conferences.models.conference import Conference
+
+logger = logging.getLogger(__name__)
+
+
+@strawberry.type
+class UpdateAttendeeTicketError:
+    message: str = ""
+
+
+@strawberry.type
+class TicketReassigned:
+    id: strawberry.ID
+    message: str = "Ticket was Successfully reassigned to {email}"
+
+
+UpdateAttendeeTicketResult = strawberry.union(
+    "UpdateAttendeeTicketResult",
+    (TicketReassigned, AttendeeTicket, UpdateAttendeeTicketError),
+)
+
+
+@strawberry.type
+class AttendeeTicketMutation:
+    @strawberry.mutation(permission_classes=[IsAuthenticated])
+    def update_attendee_ticket(
+        self, info: Info, conference: str, input: UpdateAttendeeTicketInput
+    ) -> UpdateAttendeeTicketResult:
+        conference = Conference.objects.get(code=conference)
+
+        try:
+            pretix.update_ticket(conference, input)
+
+            # TODO: filter by orderposition
+            tickets = get_user_tickets(
+                conference, info.context.request.user.email, language="en"
+            )
+
+            tickets = list(filter(lambda ticket: str(ticket.id) == input.id, tickets))
+            if tickets:
+                return tickets[0]
+
+        except Exception as e:
```

Comment:
I don't like this generic exception

I would feel more comfortable with a specific one, what error do you want to catch?

Reply:
What if the pretix api is down?

Reply:
it is not just about pretix, with `except Exception` you are catching a lot more, even code errors like:

```
>>> try:
...     5/0
... except Exception:
...     print("hello")
... 
hello
```

Reply:
I am ok with the API returning 500 or unhandled errors in graphql if something un expected happens